server/apps/research/serializers/article_serializer.py

```
@@ -9,74 +10,181 @@
-            'is_sponsored', 'sponsor_color', 'sponsor_text_color'
-        ]
+        model = RelatedArticle
+        fields = ['id', 'title', 'slug', 'thumb']
+class ArticleSerializer(serializers.ModelSerializer):
+    def create(self, validated_data):
+        try:
+            # Your existing creation logic here
+            return super().create(validated_data)
+        except DjangoValidationError as e:
+            # Handle validation errors specifically, these are safe to expose
+            raise serializers.ValidationError(e.message_dict)
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)